showcase/tests/integration/components/hds/form/index-test.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'showcase/tests/helpers';
+import { render, setupOnerror } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+import { AVAILABLE_TAGS } from '@hashicorp/design-system-components/components/hds/form/index';
+
+module('Integration | Component | hds/form/index', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it should render the component with a CSS class that matches the component name', async function (assert) {
+    await render(hbs`<Hds::Form id="test-form" />`);
+    assert.dom('#test-form').hasClass('hds-form');
+  });
+
+  // OPTIONS
+
+  // Tag
+
+  test('it should render the component using a form tag by default', async function (assert) {
+    await render(hbs`<Hds::Form id="test-form-component" />`);
+    assert.dom('#test-form-component').hasTagName('form');
+  });
+
+  test('it should render the component using a div tag if specified in the @tag prop', async function (assert) {
+    await render(hbs`<Hds::Form id="test-form-component" @tag="div" />`);
+    assert.dom('#test-form-component').hasTagName('div');
+  });
+
+  // SectionMaxWidth
+
+  test('it should set an inline style for the section max-width custom property', async function (assert) {
+    await render(
+      hbs`<Hds::Form id="test-form-component" @sectionMaxWidth="32em" />`,
+    );
+    assert.dom('#test-form-component').hasStyle(
+      {
+        '--hds-form-section-max-width': '32em',
+      },
+      'Inline style for section max-width is set',
+    );
+  });
+
+  // CONTENT
+
+  test('it should yield the different subcomponents as children, for the different available tags', async function (assert) {
+    for (const tag of AVAILABLE_TAGS) {
+      this.set('tag', tag);
+      await render(
+        hbs`<Hds::Form id="test-form" @tag={{this.tag}} as |FORM|>
+              <FORM.Header>
+                <FORM.HeaderTitle />
+                <FORM.HeaderDescription />
+              </FORM.Header>
+              <FORM.Section>
+                <FORM.SectionHeader>
+                  <FORM.SectionHeaderTitle />
+                  <FORM.SectionHeaderDescription />
+                </FORM.SectionHeader>
+                <FORM.SectionMultiFieldGroup as |FG|>
+                  <FG.Item />
+                </FORM.SectionMultiFieldGroup>
+              </FORM.Section>
+              <FORM.Separator />
+              <Hds::ButtonSet />
```

Comment:
[todo] here we would be just testing the `{{yield}}` but we already know it works from the other tests
```suggestion
```